components/brave_today/browser/feed_controller.cc

```diff
@@ -222,6 +240,40 @@ void FeedController::OnPublishersUpdated(PublishersController* controller) {
   EnsureFeedIsUpdating();
 }
 
+void FeedController::FetchCombinedFeed(GetFeedItemsCallback callback) {
+  // Handle the response
+  auto response_handler = base::BindOnce(
+      [](FeedController* controller, GetFeedItemsCallback callback, int status,
+         const std::string& body,
+         const base::flat_map<std::string, std::string>& headers) {
+        std::string etag;
+        if (headers.contains(kEtagHeaderKey)) {
+          etag = headers.at(kEtagHeaderKey);
+        }
+        VLOG(1) << "Downloaded feed, status: " << status << " etag: " << etag;
+        // Handle bad response
+        if (status < 200 || status >= 300) {
```

Comment:
Perhaps we could simplify this to `status == 200`. There's not much point in attempting to parse the body of a `204` :)

Reply:
fair enough, there's no partial content here either! done.